### PR TITLE
Balloonening ft. hyposprays + old proc inline

### PR DESCRIPTION
**Pull request by BlueHNT**
## About The Pull Request
Switches a lot of chat notices like injecting someone and drawing from a container to balloon notices. [only hypospray related]
## Why It's Good For The Game
Balloons show you that you did a thing.
## Changelog
:cl:
code: unnecessary proc/label
code: replace to_chat with balloons
/:cl:

### CODE REVIEW

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 1]
```diff
@@ -32,19 +32,10 @@
 	if(tgui_alert(user, "Are you sure you want to empty [src]?", "Flush [src]:", list("Yes", "No")) != "Yes")
 		return
 	if(isturf(user.loc))
-		to_chat(user, span_notice("You flush the contents of [src]."))
+		user.balloon_alert(user, "You flush the contents of [src].")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:35]:
```suggestion
		balloon_alert(user, "Flushes")
```
Try to keep balloon alerts as small as possible, try not to insert names like [src] unless absolutely necessary.

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 2]
```diff
@@ -54,15 +45,15 @@
 	if(istype(A, /obj/item/storage/pill_bottle) && is_open_container()) //this should only run if its a pillbottle
 		var/obj/item/storage/pill_bottle/bottle = A
 		if(reagents.total_volume >= volume)
-			to_chat(user, span_warning("[src] is full."))
+			user.balloon_alert(user, "[src] is full.")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:48]:
```suggestion
			balloon_alert(user, "Can't, full.")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 3]
```diff
@@ -54,15 +45,15 @@
...
 			return  //early returning if its full
 
 		if(!length(bottle.contents))
 			return //early returning if its empty
 		var/obj/item/pill = bottle.contents[1]
 
 		if((pill.reagents.total_volume + reagents.total_volume) > volume)
-			to_chat(user, span_warning("[src] cannot hold that much more."))
+			user.balloon_alert(user, "[src] cannot hold that much more.")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:56]:
```suggestion
			balloon_alert(user, "Can't hold that much")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 4]
```diff
@@ -77,10 +68,10 @@
 		return
 
 	if(!reagents.total_volume)
-		to_chat(user, span_warning("[src] is empty."))
+		user.balloon_alert(user, "[src] is empty.")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:71]:
```suggestion
		balloon_alert(user, "Empty")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 5]
```diff
@@ -77,10 +68,10 @@
...
 		return
 	if(!A.is_injectable() && !ismob(A))
-		to_chat(user, span_warning("You cannot directly fill [A]."))
+		user.balloon_alert(user, "You cannot directly fill [A].")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:74]:
```suggestion
		balloon_alert(user, "Can't fill [A].")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 6]
```diff
@@ -107,8 +98,8 @@
 
 	if(ismob(A))
 		var/mob/M = A
-		to_chat(user, "[span_notice("You inject [M] with [src]")].")
-		to_chat(M, span_warning("You feel a tiny prick!"))
+		user.balloon_alert(user, "You inject [M] with [src]")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:101]:
```suggestion
		balloon_alert(user, "Injects")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 7]
```diff
@@ -134,15 +125,15 @@
 	if(!A.reagents)
 		return FALSE
 	if(reagents.holder_full())
-		to_chat(user, span_warning("[src] is full."))
+		user.balloon_alert(user, "[src] is full.")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:128]:
```suggestion
		balloon_alert(user, "Can't, full")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 8]
```diff
@@ -134,15 +125,15 @@
...
 		inject_mode = HYPOSPRAY_INJECT_MODE_INJECT
 		update_icon() //So we now display as Inject
 		return FALSE
 	if(!A.reagents.total_volume)
-		to_chat(user, "<span class='warning'>[A] is empty.")
+		user.balloon_alert(user, "<span class='warning'>[A] is empty.")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:133]:
```suggestion
		balloon_alert(user, "Empty")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 9]
```diff
@@ -134,15 +125,15 @@
...
 		return
 	if(!A.is_drawable())
-		to_chat(user, span_warning("You cannot directly remove reagents from this object."))
+		user.balloon_alert(user, "You cannot directly remove reagents from this object.")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:136]:
```suggestion
		balloon_alert(user, "Can't remove reagents.")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 10]
```diff
@@ -158,18 +149,18 @@
 	var/amount = min(reagents.maximum_volume - reagents.total_volume, amount_per_transfer_from_this)
 	var/mob/living/carbon/C = A
 	if(C.get_blood_id() && reagents.has_reagent(C.get_blood_id()))
-		to_chat(user, span_warning("There is already a blood sample in [src]."))
+		user.balloon_alert(user, "There is already a blood sample in [src].")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:152]:
```suggestion
		balloon_alert(user, "Already have a blood sample")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 11]
```diff
@@ -158,18 +149,18 @@
...
 		return
 	if(!C.blood_type)
-		to_chat(user, span_warning("You are unable to locate any blood."))
+		user.balloon_alert(user, "You are unable to locate any blood.")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:155]:
```suggestion
		balloon_alert(user, "Can't locate blood")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 12]
```diff
@@ -158,18 +149,18 @@
...
 		return
 	if(C.blood_volume <= BLOOD_VOLUME_SURVIVE)
-		to_chat(user, span_warning("This body doesn't have enough blood to draw from."))
+		user.balloon_alert(user, "This body doesn't have enough blood to draw from.")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:158]:
```suggestion
		balloon_alert(user, "No blood to draw")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 13]
```diff
@@ -158,18 +149,18 @@
...
 		return
 	if(ishuman(C))
 		var/mob/living/carbon/human/H = C
 		if(H.species.species_flags & NO_BLOOD)
-			to_chat(user, span_warning("You are unable to locate any blood."))
+			user.balloon_alert(user, "You are unable to locate any blood.")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:163]:
```suggestion
			balloon_alert(user, "Can't locate blood.")
```

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 14]
```diff
@@ -183,7 +174,7 @@
 ///Checks if a container is drawable, then draw reagents from the container
 /obj/item/reagent_containers/hypospray/proc/draw_reagent(atom/A, mob/living/user)
 	var/trans = A.reagents.trans_to(src, amount_per_transfer_from_this)
-	to_chat(user, span_notice("You fill [src] with [trans] units of the solution."))
+	user.balloon_alert(user, "You fill [src] with [trans] units of the solution.")
```
Comment (RipGrayson) [inline code/game/objects/items/reagent_containers/hypospray.dm:177]:
```suggestion
	balloon_alert(user, "Fills with [trans] units.")
```

File: code/game/objects/items/reagent_containers/hypospray.dm (hunk 1 above)
Comment (BlueHNT) [reply code/game/objects/items/reagent_containers/hypospray.dm:35]: breaks code

File: code/game/objects/items/reagent_containers/hypospray.dm (hunk 2 above)
Comment (BlueHNT) [reply code/game/objects/items/reagent_containers/hypospray.dm:48]: breaks code

File: code/game/objects/items/reagent_containers/hypospray.dm (hunk 3 above)
Comment (BlueHNT) [reply code/game/objects/items/reagent_containers/hypospray.dm:56]: breaks code

File: code/game/objects/items/reagent_containers/hypospray.dm (hunk 5 above)
Comment (BlueHNT) [reply code/game/objects/items/reagent_containers/hypospray.dm:74]: breaks code

File: code/game/objects/items/reagent_containers/hypospray.dm (hunk 10 above)
Comment (BlueHNT) [reply code/game/objects/items/reagent_containers/hypospray.dm:152]: breaks code

File: code/game/objects/items/reagent_containers/hypospray.dm (hunk 12 above)
Comment (BlueHNT) [reply code/game/objects/items/reagent_containers/hypospray.dm:158]: breaks code

File: code/game/objects/items/reagent_containers/hypospray.dm (hunk 14 above)
Comment (BlueHNT) [reply code/game/objects/items/reagent_containers/hypospray.dm:177]: breaks code

**Comment by BlueHNT** [comment]
@RipGrayson hit me up in DMs if you make it work, but your changes currently won't work and will instead break food eating balloons.

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 15]
```diff
@@ -77,10 +68,10 @@
 		return
 
 	if(!reagents.total_volume)
-		to_chat(user, span_warning("[src] is empty."))
+		balloon_alert(user, "Hypospray is Empty.")
 		return
 	if(!A.is_injectable() && !ismob(A))
-		to_chat(user, span_warning("You cannot directly fill [A]."))
+		balloon_alert(user, "Can't fill [A].")
```
Comment (TiviPlus) [inline code/game/objects/items/reagent_containers/hypospray.dm:74]:
```suggestion
		A.balloon_alert(user, "Can't fill.")
```
Player just clicked so they know what A is

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 16]
```diff
@@ -283,13 +275,21 @@
...
 		update_icon()
 
 	else if(href_list["autolabeler"])
-		label(usr)
+		var/mob/user = usr
+		var/str = copytext(reject_bad_text(input(user,"Hypospray label text?", "Set label", "")), 1, MAX_NAME_LEN)
+		if(!length(str))
+			user.balloon_alert(user, "Invalid text.")
+			return
+		user.balloon_alert(user, "You label [src] as \"[str]\".")
```
Comment (TiviPlus) [inline code/game/objects/items/reagent_containers/hypospray.dm:290]:
```suggestion
		balloon_alert(user, str)
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Rest is good enough

**Pull request merged**